cmake/PC_gcc.cmake

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+get_filename_component(CMAKE_TOOL_DIR ${CMAKE_CURRENT_LIST_FILE} DIRECTORY)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_TOOL_DIR})
+
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+find_program(CMAKE_C_COMPILER NAMES gcc)
+find_program(CMAKE_CXX_COMPILER NAMES g++)
+find_program(CMAKE_ASM_COMPILER NAMES gcc)
+
+set(CMAKE_EXECUTABLE_SUFFIX_C   .elf)
```

Comment:
should these still be elf?

Reply:
File format is elf produce by gcc in default. 

Command:
[bash]$ file build/demos/projects/PC/linux/iot-middleware-sample.elf

build/demos/projects/PC/linux/iot-middleware-sample.elf: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=9e39e3316418d874b2c5b041bb2afcb9db179348, with debug_info, not stripped
h